MekHQ/src/mekhq/campaign/stratcon/StratconRulesManager.java

```diff
@@ -1679,41 +1682,46 @@ public static ReinforcementResultsType processReinforcementDeployment(Force forc
+     * file is then deserialized into a {@link ScenarioTemplate} object.</p>
+     *
+     * <p><strong>Special Cases:</strong></p>
      * <ul>
-     *   <li>If the primary unit type is `CONV_FIGHTER` or `AEROSPACEFIGHTER` (and a random check passes),
-     *       a "Low-Atmosphere" template is selected.</li>
-     *   <li>If the primary unit type qualifies as an `AEROSPACEFIGHTER` or higher,
-     *       a "Space" template is selected.</li>
-     *   <li>Otherwise, the default template is used.</li>
+     *     <li>A "Space" template is chosen if all units are aerospace and a random condition is met
+     *             (1 in 3 chance).</li>
+     *     <li>A "Low-Atmosphere" template is selected if the {@link Force} contains only airborne
+     *             units but does not meet the criteria for a "Space" template.</li>
+     *     <li>A default ground template is selected if no specific cases are matched.</li>
      * </ul>
      *
      * @param force    The {@link Force} instance that the scenario is based on.
-     *                 This is used to determine the primary unit type.
-     * @param campaign The {@link Campaign} in which the interception is taking place.
-     *                 Provides context for evaluating the {@link Force}.
-     * @return A {@link ScenarioTemplate} instance based on the template file matching the logic above,
-     *         or a default template if no specific case is matched.
-     * @see ScenarioTemplate#Deserialize(String)
-     */
-    private static ScenarioTemplate getInterceptionScenarioTemplate(Force force, Campaign campaign) {
+     *                 The force composition is used to determine the appropriate scenario template.
+     * @param hangar   The {@link Hangar} instance from which to retrieve the {@link Unit}.
+     * @return A {@link ScenarioTemplate} instance representing the chosen scenario template file based
+     *         on the logic described, or a default template if no special conditions are satisfied.
+     */
+    private static ScenarioTemplate getInterceptionScenarioTemplate(Force force, Hangar hangar) {
         String templateString = "data/scenariotemplates/%sReinforcements Intercepted.xml";
 
         ScenarioTemplate scenarioTemplate = ScenarioTemplate.Deserialize(String.format(templateString, ""));
 
-        int primaryUnitType = force.getPrimaryUnitType(campaign);
+        boolean airborneOnly = force.forceContainsOnlyAerialForces(hangar, false,
+              false);
 
-        if ((primaryUnitType == CONV_FIGHTER)
-            || (primaryUnitType == AEROSPACEFIGHTER) && (randomInt(3) == 0)) {
-            scenarioTemplate = ScenarioTemplate.Deserialize(String.format(templateString, "Low-Atmosphere "));
-        } else if (primaryUnitType >= AEROSPACEFIGHTER) {
+        boolean aerospaceOnly = false;
+        if (airborneOnly) {
+            aerospaceOnly = force.forceContainsOnlyAerialForces(hangar, false,
+                  false);
```

Comment:
This should probably also be `hanger, false, true` if I'm reading the intent and function correctly.